### PR TITLE
SLING-9974: add nullability annotations

### PR DESCRIPTION
**Pull request by karlpauls**

### CODE REVIEW

**Review by anchela — COMMENTED** [review]
looks good to me in general. i had a few comments in the acl-area where i felt that notnull would be better.

File: src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/AccessControlEntry.java [hunk 1]
```diff
@@ -34,31 +36,31 @@
...
         }
     }
 
-    public String getOperation() {
+    public @NotNull String getOperation() {
         return isAllow ? "allow" : "deny";
     }
 
-    public String getPrivileges() {
+    public @Nullable String getPrivileges() {
```
[the remark this answers is not in the document]
Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/AccessControlEntry.java:55]: Yeah, see below.

File: src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/AccessControlEntry.java [hunk 2]
```diff
@@ -34,31 +36,31 @@
 
     private final List<String> restrictions = new LinkedList<>();
 
-    public AccessControlEntry(boolean isAllow, String privileges, RepoPath repositoryPath) {
+    public AccessControlEntry(boolean isAllow, @Nullable String privileges, @NotNull RepoPath repositoryPath) {
```
[the remark this answers is not in the document]
Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/AccessControlEntry.java:39]: Right, that was the reason I didn't change it but I guess I agree - seems cleaner to just fail in that case.

File: src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/DefaultAclManager.java [hunk 3]
```diff
@@ -268,7 +263,7 @@ private static void addAclStatement(Formatter formatter, String systemUser, List
         formatter.format("end%n");
     }
 
-    private static boolean areEmpty(List<AccessControlEntry> authorizations) {
+    private static boolean areEmpty(@Nullable List<AccessControlEntry> authorizations) {
         return authorizations == null || authorizations.isEmpty();
```
[the remark this answers is not in the document]
Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/DefaultAclManager.java:267]: See above.

File: src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/DefaultAclManager.java [hunk 4]
```diff
@@ -268,7 +263,7 @@ private static void addAclStatement(Formatter formatter, String systemUser, List
         formatter.format("end%n");
     }
 
-    private static boolean areEmpty(List<AccessControlEntry> authorizations) {
+    private static boolean areEmpty(@Nullable List<AccessControlEntry> authorizations) {
```
[the remark this answers is not in the document]
Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/DefaultAclManager.java:266]: Yup, we should drop the method.

**Pull request merged**